README.md

```diff
@@ -453,6 +457,108 @@ func FunctionReconciler(c reconcilers.Config) *reconcilers.ResourceReconciler[*b
+
+**Example:**
+
+An IfThen can be used to gate a capability of the reconciler only for a resource that opts-in to the behavior. 
+
+```go
+func GatedReconciler() *reconcilers.SubReconciler[*buildv1alpha1.Function] {
+	return &reconcilers.IfThen[*buildv1alpha1.Function]{
+		If: func(ctx context.Context, resource *buildv1alpha1.Function) bool {
+			return resource.Labels["capability-gate.example/enabled"] == "true"
+		}
+		Then: reconcilers.Sequence[*buildv1alpha1.Function]{
+			// use the gated feature
+		},
+	}
+}
+```
+
+#### While
+
+A [`While`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#While) calls the reconciler so long as the condition is true, up to the maximum number of iterations (defaults to 100). The current iteration index can be retried with [`RetrieveIteration`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#RetrieveIteration).
+
+This reconciler must not be used to wait for an external to change, or for polling as this will block the reconciler queue. It to return with the result requesting to be requeued, or to watch the external state for changes that enqueue the reconcile request.
+
+**Example:**
+
+An While can be used to fan out. 
+
+```go
+func TenTimesReconciler() *reconcilers.SubReconciler[*buildv1alpha1.Function] {
+	return &reconcilers.While[*buildv1alpha1.Function]{
+		Condition: func(ctx context.Context, resource *buildv1alpha1.Function) bool {
+			return reconcilers.RetrieveIteration(ctx) < 10
+		}
+		Reconciler: reconcilers.SyncReconciler[*buildv1alpha1.Function]{
+			Sync: func(ctx context.Context, resource *buildv1alpha1.Function) error {
+				// called ten times
+				return nil
+			}
+		},
+	}
+}
+```
+
+#### TryCatch
+
+A [`TryCatch`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#TryCatch) is used to recover from errors returned by a reconciler. The `Catch` method is called with the result and error from the `Try` reconciler, giving it the option to either continue the existing results, or replace them with new results.
+
+The `Finally` reconciler is always called before returning, but does not alter the results unless it errors. The finally reconciler should avoid complex logic and be limited to cleaning up common state from the try reconciler.
+
+**Example:**
+
+An While can be used to fan out. 
+
+```go
+func IgnoreErrorsReconciler() *reconcilers.SubReconciler[*buildv1alpha1.Function] {
+	return &reconcilers.TryCatch[*buildv1alpha1.Function]{
+		Try: &reconcilers.SyncReconciler[*buildv1alpha1.Function]{
+			Sync: func(ctx context.Context, resource *buildv1alpha1.Function) error {
+				return fmt.Errorf("alway error?")
+			}
+		},
+		Catch: func(ctx context.Context, resource *buildv1alpha1.Function, result reconcile.Result, err error) (reconcile.Result,  error) {
+			// suppress error
```

Comment:
```suggestion
			// suppress error and/or
        // log it and/or
        // update the parent's status
```

Reply:
in this example, it's surprising the error. Logging or updating status are other things that could be done.